src/modules/iam/nm_iam_internal.c

```diff
@@ -170,6 +170,10 @@ struct nm_iam_user* nm_iam_internal_pair_new_client(struct nm_iam* iam, NabtoDev
         return NULL;
     }
 
+    if (strlen(sct) > iam->sctMaxLength) {
+        return NULL;
```

Comment:
We are leaking the fingerprint and SCT here, probably because that bug is already in the code above this.